Sources/ZcashLightClientKit/Rust/ZcashRustBackend.swift

```diff
@@ -408,392 +293,420 @@ class ZcashRustBackend: ZcashRustBackendWelding {
+        case 0: //ok
+            return DbInitResult.success
+        case 1:
+            return DbInitResult.seedRequired
+        default:
+            if let error = lastError() {
+                throw throwDataDbError(error)
+            } else {
+                throw RustWeldingError.dataDbInitFailed(message: "Database initialization failed: unknown error")
+            }
+        }
+    }
+    
+    static func isValidSaplingAddress(_ address: String, networkType: NetworkType) throws -> Bool {
+        guard !address.containsCStringNullBytesBeforeStringEnding() else {
+            return false
+        }
         
-        return zcashlc_shield_funds(
-            dbData.0,
-            dbData.1,
-            account,
-            [CChar](xprv.utf8CString),
-            memo.bytes,
-            spendParamsPath,
-            UInt(spendParamsPath.lengthOfBytes(using: .utf8)),
-            outputParamsPath,
-            UInt(outputParamsPath.lengthOfBytes(using: .utf8)),
-            networkType.networkId
-        )
+        guard zcashlc_is_valid_shielded_address([CChar](address.utf8CString), networkType.networkId) else {
+            if let error = lastError() {
+                throw error
+            }
+            return false
+        }
+
+        return true
+    }
+    
+    static func isValidTransparentAddress(_ address: String, networkType: NetworkType) throws -> Bool {
+        guard !address.containsCStringNullBytesBeforeStringEnding() else {
+            return false
+        }
+        
+        guard zcashlc_is_valid_transparent_address([CChar](address.utf8CString), networkType.networkId ) else {
+            if let error = lastError() {
+                throw error
+            }
+            return false
+        }
+
+        return true
     }
     
-    static func deriveSaplingExtendedFullViewingKey(_ spendingKey: SaplingExtendedSpendingKey, networkType: NetworkType) throws -> SaplingExtendedFullViewingKey? {
-        guard !spendingKey.stringEncoded.containsCStringNullBytesBeforeStringEnding() else {
-            throw RustWeldingError.malformedStringInput
+    static func isValidSaplingExtendedFullViewingKey(_ key: String, networkType: NetworkType) throws -> Bool {
+        guard !key.containsCStringNullBytesBeforeStringEnding() else {
+            return false
         }
         
-        guard
-            let extfvk = zcashlc_derive_extended_full_viewing_key(
-                [CChar](spendingKey.stringEncoded.utf8CString),
-                networkType.networkId
-            )
-        else {
+        guard zcashlc_is_valid_viewing_key([CChar](key.utf8CString), networkType.networkId) else {
             if let error = lastError() {
                 throw error
             }
-            return nil
+            return false
         }
 
-        defer { zcashlc_string_free(extfvk) }
+        return true
```

Comment:
nit: I think that this would read more clearly as:
```suggestion
        let result = zcashlc_is_valid_viewing_key([CChar](key.utf8CString), networkType.networkId)
        if let error = lastError() {
            throw error
        } else {
            return result
        }
```